dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUI.cpp

```diff
@@ -0,0 +1,137 @@
+        auto tempFilePath = tempFile.Path();
+        auto token = SharedStorageAccessManager::AddFile(tempFile);
+
+        co_return token_and_path{ token, tempFilePath };
+    }
+
+    IAsyncOperation<StorageFile> CameraCaptureUI::CaptureFileAsync(CameraCaptureUIMode mode)
+    {
+        // Only allow one capture operation at a time from this instance
+        std::unique_lock<std::mutex> lock(m_captureMutex, std::try_to_lock);
+        if (!lock.owns_lock())
+        {
+            throw hresult_error(E_ILLEGAL_METHOD_CALL, L"CameraCaptureUI is already capturing");
+        }
+
+        bool isAppPackaged = m_telemetryHelper.IsPackagedApp();
+        PCWSTR appName = m_telemetryHelper.GetAppName().c_str();
+        const wchar_t* mediaType = nullptr;
+        try
+        {
+            auto strong = get_strong();
+
+            if (mode == CameraCaptureUIMode::PhotoOrVideo)
+            {
+                mediaType = L"photoOrVideo";
+            }
+            else if (mode == CameraCaptureUIMode::Photo)
+            {
+                mediaType = L"photo";
+            }
+            else if (mode == CameraCaptureUIMode::Video)
+            {
+                mediaType = L"video";
+            }
+            else
+            {
+                throw hresult_invalid_argument();
+            }
+            auto logCaptureOperation{ CameraCaptureUITelemetry::CameraCaptureOperation::Start(isAppPackaged, appName, mediaType) };
+
+            ValueSet properties;
+            properties.Insert(L"MediaType", box_value(mediaType));
+
+            if ((mode == CameraCaptureUIMode::PhotoOrVideo) || (mode == CameraCaptureUIMode::Photo))
+            {
+                m_photoSettings->validate();
+                m_photoSettings->Serialize(properties);
+                m_photoTokenFile = co_await CreateEmptyFileAndGetToken(m_photoSettings->GetFileExtension());
+                properties.Insert(L"PhotoFileToken", box_value(m_photoTokenFile.token));
+            }
+
+            if ((mode == CameraCaptureUIMode::PhotoOrVideo) || (mode == CameraCaptureUIMode::Video))
+            {
+                m_videoSettings->validate();
+                m_videoSettings->Serialize(properties);
+                m_videoTokenFile = co_await CreateEmptyFileAndGetToken(m_videoSettings->GetFileExtension());
+                properties.Insert(L"VideoFileToken", box_value(m_videoTokenFile.token));
+            }
+
+            auto tokenValue = co_await LaunchCameraForResultToken(m_windowId, properties);
+            if (tokenValue.empty())
+            {
+                co_return nullptr;
+            }
+            StorageFile file = nullptr;
+
+            file = co_await SharedStorageAccessManager::RedeemTokenForFileAsync(tokenValue);
+            logCaptureOperation.Stop();
+            co_return file;
+        }
+        catch (...)
+        {
+            auto e = winrt::hresult_error(winrt::to_hresult(), winrt::take_ownership_from_abi);
```

Comment:
Do you need to do this? The default WIL activity system automatically grabs and logs thrown exceptions at point of failure. You shouldn't need this try/catch block here - just start the telemetry activity above, and errors will flow through the normal WIL->Telemetry provider pipeline.

Reply:
Updated!

I wanted to make sure I get back the error telemetry for creating measures. I figured, the job gets done with the Telemetry activity provider pipeline as well. Got rid of the try/catch, relying on the telemetry activity now!